src/hearings/containers/hearing-actuals/hearing-actuals-view-edit-parties/hearing-actuals-view-edit-parties.component.ts

```diff
@@ -0,0 +1,152 @@
+    this.hearingRole = this.route.snapshot.data.hearingRole;
+    this.sub = this.hearingStore.select(fromHearingStore.getHearingActuals)
+      .pipe(
+        filter((state: HearingActualsStateData) => !!state.hearingActualsMainModel)
+      )
+      .subscribe((state: HearingActualsStateData) => {
+        // create new deep clone otherwise cannot modify model
+        this.hearingActuals = JSON.parse(JSON.stringify(state.hearingActualsMainModel));
+        this.createForm(this.hearingActuals);
+      });
+  }
+
+  public get parties(): FormArray {
+    return this.partiesTable.get('parties') as FormArray;
+  }
+
+  public initiateForm(): FormGroup {
+    return this.fb.group({
+      firstName: [''],
+      lastName: [''],
+      role: ['Please select'],
+      attendanceType: ['Please select'],
+      organisation: [''],
+      attendeeRepresenting: ['Please select'],
+      isParty: [false]
+    });
+  }
+
+  public addRow($event: Event): void {
+    $event.preventDefault();
+    ($event.target as HTMLElement).blur();
+    this.parties.push(this.initiateForm());
+    setTimeout(() => {
+      this.renderer.selectRootElement('tr:last-child input').focus();
+    }, 100);
+  }
+
+  public removeRow($event: Event, index: number): void {
+    $event.preventDefault();
+    this.parties.removeAt(index);
+  }
+
+  public ngOnDestroy() {
+    this.sub.unsubscribe();
+  }
+
+  private createForm(hearingActuals: HearingActualsMainModel): void {
+
+    hearingActuals.hearingPlanned.plannedHearingDays[0].parties.forEach( party => {
+
+      this.participants.push({
+        name: `${party.individualDetails.firstName} ${party.individualDetails.lastName}`,
+        id: party.partyId
+      });
+
+      this.parties.push(this.fb.group({
+        firstName: [party.individualDetails.firstName],
+        lastName: [party.individualDetails.lastName],
+        role: [party.partyRole],
+        attendanceType: [party.partyChannelSubType],
+        organisation: [party.organisationDetails.name],
+        attendeeRepresenting: [party.partyId],
+        isParty: [true]
+      }));
+    });
+
+    hearingActuals.hearingActuals.actualHearingDays.forEach( actualHearingDay => {
+
+      actualHearingDay.actualDayParties.forEach( dayParty => {
+        if (dayParty.didNotAttendFlag) {
+          dayParty.partyChannelSubType = 'notAttending';
+        }
+
+        this.parties.push(this.fb.group({
+          firstName: [dayParty.actualIndividualDetails.firstName],
+          lastName: [dayParty.actualIndividualDetails.lastName],
+          role: [dayParty.partyRole],
+          attendanceType: [dayParty.partyChannelSubType],
+          organisation: [dayParty.actualOrganisationDetails.name],
+          attendeeRepresenting: [dayParty.representedParty],
+          isParty: [false]
+        }));
+      });
+    });
+
+  }
+
+  public submitForm($event): void {
+    $event.preventDefault();
+    // console.log(this.parties.value);
```

Comment:
could we remove this line?

Reply:
yeah sure, only left it there so Kuda could pick up the values easily when posting to API etc